sktime/forecasting/pytorchforecasting.py

```diff
@@ -0,0 +1,348 @@
+                patience=3,
+                verbose=False,
+                mode="min",
+            )
+            params = [
+                {
+                    "trainer_params": {
+                        "max_epochs": 1,  # for quick test
+                    },
+                    "dataset_params": {
+                        "max_encoder_length": 3,
+                    },
+                },
+                {
+                    "trainer_params": {
+                        "callbacks": [early_stop_callback],
+                    },
+                    "model_params": {
+                        "hidden_size": 10,
+                        "dropout": 0.1,
+                        # "loss": QuantileLoss(),
+                        # can not pass test_set_params and test_set_params_sklearn
+                        # QuantileLoss() != QuantileLoss()
+                        "optimizer": "Adam",
+                        # avoid jdb78/pytorch-forecasting#1571 bug in the CI
+                        "log_val_interval": -1,
+                    },
+                    "dataset_params": {
+                        "max_encoder_length": 3,
+                    },
+                },
+            ]
+
+        return params
+
+
+class PytorchForecastingNBeats(_PytorchForecastingAdapter):
+    """pytorch-forecasting NBeats model.
+
+    Parameters
+    ----------
+    model_params :  Dict[str, Any] (default=None)
+        parameters to be passed to initialize the pytorch-forecasting NBeats model [1]_
+        for example: {"num_blocks": [5, 5], "widths": [128, 1024]}
+    dataset_params : Dict[str, Any] (default=None)
+        parameters to initialize `TimeSeriesDataSet` [2]_ from `pandas.DataFrame`
+        max_prediction_length will be overwrite according to fh
+        time_idx, target, group_ids, time_varying_known_reals, time_varying_unknown_reals
+        will be infered from data, so you do not have to pass them
+    train_to_dataloader_params : Dict[str, Any] (default=None)
+        parameters to be passed for `TimeSeriesDataSet.to_dataloader()`
+        by default {"train": True}
+    validation_to_dataloader_params : Dict[str, Any] (default=None)
+        parameters to be passed for `TimeSeriesDataSet.to_dataloader()`
+        by default {"train": False}
+
+    References
+    ----------
+    .. [1] https://pytorch-forecasting.readthedocs.io/en/stable/api/pytorch_forecasting.models.nbeats.NBeats.html
+    .. [2] https://pytorch-forecasting.readthedocs.io/en/stable/api/pytorch_forecasting.data.timeseries.TimeSeriesDataSet.html
+    """  # noqa: E501
+
+    _tags = {
+        # packaging info
+        # --------------
+        # "authors": ["XinyuWu"],
+        # "maintainers": ["XinyuWu"],
+        # "python_dependencies": "pytorch_forecasting"
+        # inherited from _PytorchForecastingAdapter
+        # estimator type
+        # --------------
+        "python_dependencies": ["pytorch_forecasting>=1.0.0", "torch", "lightning"],
```

Comment:
Is lightning and torch not directly installed via PyTorch_forecasting?

Reply:
Yes, it's automatically installed in my enviroment. However, the [docs](https://pytorch-forecasting.readthedocs.io/en/stable/) of pytorch-forecasting are indicating other possibilities, so I add it for sure.

Reply:
I See